packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.mm

```diff
@@ -234,7 +234,9 @@ - (BOOL)textView:(__unused UITextView *)textView shouldChangeTextInRange:(NSRang
     return NO;
   }
 
-  if ([newText isEqualToString:text]) {
+  if (range.location + range.length > _backedTextInputView.text.length) {
+    range = NSMakeRange(range.location, _backedTextInputView.text.length - range.location);
+  } else if ([newText isEqualToString:text]) {
```

Comment:
Shouldn't it be something like this?

```suggestion
  if (range.location + range.length > _backedTextInputView.text.length) {
    range = NSMakeRange(range.location, _backedTextInputView.text.length - range.location);
  } 
  
  if ([newText isEqualToString:text]) {
```

I'm not super familiar with the internals, but in the first branch, we are updating the `range` and the second branch don't do anything related to range, so it feels weird that we have a logic that is:

```
// if range is greater that the backed text length
// then do something
// else if the text has not changed, 
// then do something else.
``` 

the value of range and the comparison between `newText` and `text` are unrelated no?

Reply:
@cipolleschi Yes, it works with both architectures. 
The reason is if the range is bigger than text input's text range, we can't check whether the text changed, it may return `YES` to let the system change the text input and lead to crash, instead we should skip the text equal check and replace the text input by ourself.